test/installation.sh

```diff
@@ -0,0 +1,184 @@
+: '
+ * Copyright (C) 2024 The pgmoneta community
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list
+ * of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this
+ * list of conditions and the following disclaimer in the documentation and/or other
+ * materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without specific
+ * prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *'
+
+#!/bin/bash
+
+install_dependencies_debian() {
+  export DEBIAN_FRONTEND=noninteractive
+  sudo apt-get update
+  sudo apt-get install -y --no-install-recommends git gcc cmake make wget build-essential libreadline-dev zlib1g-dev flex bison libxml2-dev libxslt-dev libssl-dev libxml2-utils xsltproc ccache pkg-config
+}
+
+install_dependencies_redhat() {
+  sudo yum install -y git gcc cmake make bison-devel readline-devel zlib-devel openssl-devel wget ccache libicu-devel flex libxml2-devel libxslt-devel perl
+  sudo yum groupinstall -y 'Development Tools'
+}
+
+check_postgresql_installed() {
+  if command -v pg_config >/dev/null 2>&1; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+get_latest_patch_version() {
+  version=${1:-13}
+  url="https://ftp.postgresql.org/pub/source/"
+  if [ "$version" -eq 17 ]; then
+    latest_version=$(curl -s $url | grep -oP "17beta1" | sort -V | tail -1)
+  else
+    latest_version=$(curl -s $url | grep -oP "${version}\.\d+" | sort -V | tail -1)
+  fi
+  REPLY=$latest_version
+}
+
+install_postgresql() {
+  version=${1:-13.0}
```

Comment:
13.0

Reply:
This is a default value to be used in case the version value is empty. It is just for safety considerations and will not be used, as we will obtain the latest patch release number from the `get_latest_patch_version` function.

Reply:
I think it would be best to compile from `REL_XX_STABLE`, then install the extension after.

Relying on repositories may not be good in the long run.

But we can take it in steps -- but, part of this patch need to into pgmoneta/Core in order to use/verify the extension functionality - so we need to be really sure that it is what we want

Reply:
I checked the Git repository, but version 17 does not have the `REL_XX_STABLE` branch.

Reply:
We can only install versions 13 to 16 from the `REL_XX_STABLE` branch.

Reply:
`REL_17_STABLE` will likely be next week - we can add it later